crates/stackable-operator/src/crd/git_sync/v1alpha1_impl.rs

```diff
@@ -0,0 +1,798 @@
+            Some(("--depth".to_string(), git_sync.depth.to_string())),
+            Some((
+                "--period".to_string(),
+                format!("{}s", git_sync.wait.as_secs()),
+            )),
+            Some(("--link".to_string(), GIT_SYNC_LINK.to_string())),
+            Some(("--root".to_string(), GIT_SYNC_ROOT_DIR.to_string())),
+            one_time.then_some(("--one-time".to_string(), "true".to_string())),
+        ]
+        .into_iter()
+        .flatten()
+        .collect::<BTreeMap<_, _>>();
+
+        let internal_git_config = [(
+            GIT_SYNC_SAFE_DIR_OPTION.to_string(),
+            GIT_SYNC_ROOT_DIR.to_string(),
+        )]
+        .into_iter()
+        .collect::<BTreeMap<_, _>>();
+
+        let mut user_defined_args = BTreeMap::new();
+        // The key and value in Git configs are separated by a colon, but both
+        // can contain either escaped colons or unescaped colons if enclosed in
+        // quotes. To avoid parsing, just a vector is used instead of a map.
+        let mut user_defined_git_configs = Vec::new();
+
+        for (key, value) in &git_sync.git_sync_conf {
+            // The initial git-sync implementation in the airflow-operator
+            // (https://github.com/stackabletech/airflow-operator/pull/381)
+            // used this condition to find Git configs. It is also used here
+            // for backwards-compatibility:
+            if key.to_lowercase().ends_with("-git-config") {
+                // Roughly check if the user defined config contains an
+                // internally defined config and emit a warning in case.
+                if internal_git_config.keys().any(|key| value.contains(key)) {
+                    tracing::warn!("Config option {value:?} contains a value for {GIT_SYNC_SAFE_DIR_OPTION} that overrides
+                                the value of this operator. Git-sync functionality will probably not work as expected!");
+                }
+                user_defined_git_configs.push(value.to_owned());
+            } else if internal_args.contains_key(key) {
+                tracing::warn!(
+                    "The git-sync option {key:?} is already internally defined and will be ignored."
+                );
+            } else {
+                // The user-defined arguments are not validated.
+                user_defined_args.insert(key.to_owned(), value.to_owned());
+            }
+        }
+
+        // The user-defined Git config is just appended.
+        // The user is responsible for escaping special characters like `:` and `,`.
+        let git_config = internal_git_config
+            .into_iter()
+            .map(|(key, value)| format!("{key}:{value}"))
+            .chain(user_defined_git_configs)
+            .collect::<Vec<_>>()
+            .join(",");
+
+        let mut args = internal_args;
+        args.extend(user_defined_args);
+        args.insert("--git-config".to_string(), format!("'{git_config}'"));
```

Comment:
question: Any particular reason why we don't operate on the `internal_args` directly? One of the above suggestions even makes them `mut`.